mlflow/models/evaluation/base.py

```diff
@@ -0,0 +1,530 @@
+           columns except the "labels" column.
+
+        :param labels: One of the following:
+         - A numpy array or list of evaluation labels, if `data` is also a numpy array or list.
+         - The string name of a column from `data` that contains evaluation labels, if `data`
+           is a DataFrame.
+
+        :param name: (Optional) The name of the dataset (must not contain ").
+
+        :param path: (Optional) the path to a serialized DataFrame (must not contain ").
+          (e.g. a delta table, parquet file)
+        """
+        try:
+            from pyspark.sql import DataFrame as SparkDataFrame
+
+            supported_dataframe_types = (pd.DataFrame, SparkDataFrame)
+        except ImportError:
+            supported_dataframe_types = (pd.DataFrame,)
+
+        if name is not None and '"' in name:
+            raise ValueError(f'Dataset name cannot include a double quote (") but got name {name}')
+        if path is not None and '"' in path:
+            raise ValueError(f'Dataset path cannot include a double quote (") but got name {path}')
+
+        if isinstance(data, (np.ndarray, list)):
+            if not isinstance(labels, (np.ndarray, list)):
+                raise ValueError(
+                    "If data is a numpy array or list of evaluation features, "
+                    "labels must be a numpy array or list of evaluation labels"
+                )
+        elif isinstance(data, supported_dataframe_types):
+            if not isinstance(labels, str):
+                raise ValueError(
+                    "If data is a Pandas DataFrame or Spark DataFrame, labels must be the "
+                    "string name of a column from `data` that contains evaluation labels"
+                )
+        else:
+            raise ValueError(
+                "The data argument must be a numpy array, a list or a " "Pandas DataFrame."
+            )
+
+        self._user_specified_name = name
+        self.data = data
+        self.labels = labels
+        self.path = path
+        self._hash = None
+
+    def _extract_features_and_labels(self):
+        """
+        Extract features data and labels data.
+        For spark dataframe, will only extract the first SPARK_DATAFRAME_LIMIT rows data
+        and emit warning.
+        """
+        if isinstance(self.data, np.ndarray):
+            return self.data, self.labels
+        else:
+            if not isinstance(self.data, pd.DataFrame):
+                data = self.data.limit(EvaluationDataset.SPARK_DATAFRAME_LIMIT).toPandas()
+                _logger.warning(
+                    f"Only the first {EvaluationDataset.SPARK_DATAFRAME_LIMIT} rows in the "
+                    f"spark dataframe are examined."
+                )
+            else:
+                data = self.data
+            feature_cols = [x for x in data.columns if x != self.labels]
+            return data[feature_cols], data[self.labels]
```

Comment:
```suggestion
            return data.drop(self.labels, axis=1), data[self.labels]
```

does the same thing.

Reply:
The design doc has decide the labels must be a string (only one label column allowed)
but using pandas_df.drop make code more simple, I adopt it.